packages/experimental/src/components/ImportModal/ImportModal.js

```diff
@@ -0,0 +1,316 @@
+        newFile.errorBody = fetchErrorBody || defaultErrorBody;
+        newFile.errorSubject = fetchErrorHeader || errorHeader;
+        newFile.invalid = true;
+      } else if (newFile.invalidFileType) {
+        newFile.errorBody = invalidFileTypeErrorBody || defaultErrorBody;
+        newFile.errorSubject = invalidFileTypeErrorHeader || errorHeader;
+        newFile.invalid = true;
+      } else if (maxFileSize && newFile.filesize > maxFileSize) {
+        newFile.errorBody = maxFileSizeErrorBody || defaultErrorBody;
+        newFile.errorSubject = maxFileSizeErrorHeader || errorHeader;
+        newFile.invalid = true;
+      }
+      return newFile;
+    });
+    const finalFiles = [...files, ...updatedFiles];
+    setFiles(finalFiles);
+  };
+
+  const fetchFile = async (evt) => {
+    evt.preventDefault();
+    const fileName = importUrl
+      .substring(importUrl.lastIndexOf('/') + 1)
+      .split('?')[0];
+    const pendingFile = {
+      name: fileName,
+      status: 'uploading',
+      uuid: uuidv4(),
+    };
+    const updatedFiles = [...files, pendingFile];
+    setFiles(updatedFiles);
+    try {
+      const response = await fetch(importUrl);
+      if (!response.ok || response.status !== 200)
+        throw new Error(response.status);
+      const blob = await response.blob();
+      const newFile = new File([blob], fileName, { type: blob.type });
+      newFile.invalidFileType = isInvalidFileType(newFile);
+      newFile.uuid = pendingFile.uuid;
+      updateFiles([newFile]);
+    } catch (err) {
+      console.error('fetch failed', err);
+      const newFile = {
+        ...pendingFile,
+        fetchError: true,
+      };
+      updateFiles([newFile], true);
+    }
+  };
+
+  const onAddFile = (evt, { addedFiles }) => {
+    evt.stopPropagation();
+    updateFiles(addedFiles);
+  };
+
+  const onRemoveFile = (uuid) => {
+    const updatedFiles = files.filter((f) => f.uuid !== uuid);
+    setFiles(updatedFiles);
+  };
+
+  const onSubmitHandler = () => {
+    onRequestSubmit(files);
+  };
+
+  const inputHandler = (evt) => {
+    setImportUrl(evt.target.value);
+  };
+
+  const primaryButtonDisabled = !files.length || files.some((f) => f.invalid);
+  const importButtonDisabled = !importUrl;
+
+  return (
+    <Modal
+      open={open}
+      primaryButtonText={primaryButtonText}
+      secondaryButtonText={secondaryButtonText}
+      modalHeading={modalHeading}
+      primaryButtonDisabled={primaryButtonDisabled}
+      onRequestSubmit={onSubmitHandler}
+      onRequestClose={onRequestClose}
+      className={`${expPrefix}--import-modal`}>
+      <div className={`${expPrefix}--import-modal-inner`}>
+        <p className={`${expPrefix}--import-modal-body`}>{modalBody}</p>
+        <p className={`${expPrefix}--import-modal-label`}>{fileDropHeader}</p>
+        <FileUploaderDropContainer
+          accept={validFileTypes}
+          labelText={fileDropLabel}
+          onAddFiles={onAddFile}
+          multiple
```

Comment:
Should be a boolean property of the modal for added flexibility. Otherwise looks good to me.